### PR TITLE
Feature/US16492 Update Projects Model

### PR DESCRIPTION
**Pull request by vchtelmakh**
Previously, we had `shifts` with nested data as a string. This is replaced by a single `Project` model.

### CODE REVIEW

File: src/app/project-detail/project-detail.component.ts [hunk 1]
```diff
@@ -43,58 +40,30 @@ export class ProjectDetailComponent implements OnInit {
...
-  }
-
-  public analyticsTracking(projectDetail: ProjectDetail) {
+  public analyticsTracking(projectDetail: Project) {
     this.pending = true;
     this.analyticsService.trackProjectJoined(projectDetail);
     // Due to redirect, let's wait a second so that analytics call above successfully
     // completes and doesn't get cancelled when externally redirected
     setTimeout(() => {
-      window.location.href = this.volLocalUrl;
```
Comment (andycanterbury) [inline src/app/project-detail/project-detail.component.ts]: So, until we build the sign up page, with this gone, we go nowhere when they click Join Project?

File: src/app/services/filter-shifts/filter-shifts.service.spec.ts [hunk 2]
```diff
@@ -112,64 +118,64 @@ describe('FilterShiftsService', () => {
...
-      const unfilteredShifts: any[] = [
-        {shift_id : 1, job_custom_api_data: {captName: 'Angie Robey'}},
-        {shift_id : 2, job_custom_api_data: {captName: 'Sandi Ritter'}},
-        {shift_id : 3, job_custom_api_data: {captName: 'Wole Ojo'}},
-        {shift_id : 4, job_custom_api_data: {captName: 'A.J. Colubiale'}},
-        {shift_id : 5, job_custom_api_data: {captName: 'Victor Chtelmakh'}},
-        {shift_id : 6, job_custom_api_data: {captName: 'Angie Robey'}},
+      const unfilteredProjects: Project[] = [
+        {id : 1, captainName: 'Angie Robey'} as Project,
+        {id : 2, captainName: 'Sandi Ritter'} as Project,
```
Comment (andycanterbury) [inline src/app/services/filter-shifts/filter-shifts.service.spec.ts]: 😢

File: src/app/project-detail/project-detail.component.ts (hunk 1 above)
Comment (vchtelmakh) [reply src/app/project-detail/project-detail.component.ts]: Yea - I didn't see a point in taking them to golocal. I also wanted to delete the `buildGoLocalUrl` function.

File: src/app/services/filter-shifts/filter-shifts.service.spec.ts (hunk 2 above)
Comment (vchtelmakh) [reply src/app/services/filter-shifts/filter-shifts.service.spec.ts]: 😢

**Pull request merged**